### ISSUE TITLE
Having the same path alias for different next.js microsites

### ISSUE THREAD

**Issue by shaxaaa** [chapter-three/next-drupal#279]
Hi guys,

First of all, great work! We are doing a PoC if the next module is a good fit for us and everything seems to be working so far. One thing we weren't able to see though is if we can have the same alias for multiple next.js sites. For example we have 3 presentation sites and all 3 of them should have a page aliased "/contacts" but still the node is different. We are thinking of prefexing them automatically with the site id in the alias and then do a reverse change on the next.js site from translatePathFromContext.

Did someone tackled this already or is there something done out of the box?

**Comment by alex-ahumada** [comment]
We've solved a similar requirement in our project using Next.js rewrites:
https://nextjs.org/docs/api-reference/next.config.js/rewrites

**Comment by shadcn** [comment]
Yes like @alex-ahumada said, you can do this using Next.js rewrites.

Let's say you have 3 microsites on Drupal, each with their own contact page:

![Untitled-2022-03-23-1048-2-2](https://user-images.githubusercontent.com/124599/183395175-12dc7801-545e-4301-8ce8-4e3ac5e67a95.png)

In your `next.config.js`, you can use `rewrites` to map `/contact` to `/microsite-a/contact`:

```js
const nextConfig = {
  rewrites: async () => {
    return [
      {
        source: "/contact",
        destination: "/microsite-a/contact",
      },
    ]
  },
}

module.exports = nextConfig
```

With the config above, visiting `https://microsite-a.com/contact` will render the contact page for Microsite A. You can do same for the other Next.js sites.

You can also use a similar strategy to handle more pages that are shared on the Drupal site using dynamic routes:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  rewrites: async () => {
    return [
      {
        source: "/microsite-a/:path*",
        destination: "/:path*",
      },
    ]
  },
}

module.exports = nextConfig
```

This way, anything that resolves at `/microsite-a/PATH` in Drupal can be accessed at `/PATH` on your Next.js site.

Hope this helps.

**Comment by shaxaaa** [comment]
@shadcn curries if you are using pathauto for prefixing the pages with the microsites in your example and what happens afterwards when your are building your sitemaps, real time SEOs in Drupal with Yoast and etc.

One more question that I'm exploring is potentially different enabled languages per microsite and different translations per microsite.

**Comment by martdavidson** [comment]
@shaxaaa Did you explore this approach anymore? I'm about to dive into this with a multilingual multi-frontend single drupal install.

Also curious if it breaks ISR.
